durable-storage/src/persistence_layer.rs

```diff
@@ -35,6 +36,24 @@ use crate::repo::DirectoryManagerError;
 /// Type alias for a 32-byte hash used for identifying key-value blobs & commits.
 type Hash = [u8; 32];
 
+/// [`CommitId`]'s are used to generate [`PersistenceLayer`] commits & to checkout specific commits
+/// from a [`DirectoryManager`].
+#[derive(Debug, Clone, PartialEq)]
+pub struct CommitId(Hash);
+
+impl From<&Hash> for CommitId {
```

Comment:
nit/feel free to resolve 

`Hash` implements `Copy` - so there's no need to take a reference? Especially if the hash is only created once, you can just pass it